core/model/modx/processors/resource/update.class.php

```diff
@@ -539,6 +543,10 @@ public function checkWebLinkTarget() {
                 $this->addFieldError('modx-weblink-content', $this->modx->lexicon('resource_err_weblink_target_nf'));
                 return false;
             }
+
+            if ($targetResource->get('id') === $this->object->get('id')) {
+                $this->addFieldError('modx-weblink-content', $this->modx->lexicon('resource_err_weblink_target_self'));
```

Comment:
But function returns true even if error showed. Should be there `return false;`?

Reply:
You are right. When I saw your comment I was thinking, why didn't I noticed it when testing? If it would return true the resource should save but it didn't.. 
Then I checked and figured out that this method returns false if there are errors set: https://github.com/modxcms/revolution/blob/b8e8ac20122878d08bb525a51b472ea77dca3e27/core/model/modx/processors/resource/update.class.php#L167

Reply:
The modObjectUpdate processor does not go further if any field error is set during beforeSet or beforeSave. So return true/false is not really needed. https://github.com/modxcms/revolution/blob/b8e8ac20122878d08bb525a51b472ea77dca3e27/core/model/modx/modprocessor.class.php#L862, It is just used to avoid unnecessary tests.